rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -0,0 +1,319 @@
+package coop.rchain.rspace.history
+
+import cats.syntax.all._
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import scodec.Codec
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+import scala.collection.concurrent.TrieMap
+
+object RadixTree {
+
+  /**
+    * Radix Tree structure (Node and Leaf)
+    */
+  sealed trait TreeR
+  final case class NodeR(prefix: ByteVector, refs: Vector[ByteVector]) extends TreeR
+  final case class LeafR(prefix: ByteVector, value: ByteVector)        extends TreeR
```

Comment:
Are you considering `extensionNode`(extension in eth term and `Skip` in rchain term ) same as `LeafR` here?

Reply:
I'd like to be simple as possible so `Tree` with `Branch` and `Leaf` nodes seems like a good choice.

Do you know what's the meaning of _extension_ node in ETH?

Reply:
https://ethereum.stackexchange.com/questions/39915/ethereum-merkle-patricia-trie-extension-node
I think this is kind of a good answer.

Reply:
The extension node doesn't seem like a good way to resolve inserting or deleting large amount of nodes.

Reply:
Maybe some tests are better to evaluate it.